### DISCUSSION TITLE
Installing multiple nested packages

### DISCUSSION THREAD

**Discussion by rmartin16** [tox-dev/tox#3232] [Q&A] [answered]
I've been trying to work out how `tox` can install multiple packages for a project.

This is a simplified version of the src layout.
```
project
├── core
│   ├── pyproject.toml
│   ├── README.rst
│   ├── src
│   │   ├── pkg1
│   └── tests
├── docs
├── dummy
│   ├── pyproject.toml
│   ├── README.rst
│   └── src
│       ├── pkg2
├── LICENSE
├── pyproject.toml
├── README.rst
├── tox.ini
```
**Can `tox` be configured to install both the `core` and `dummy` packages?**

Currently, I'm running `pip install ./core ./dummy` as one of the commands but this creates issues for running envs in parallel because multiple envs are trying to build and install the packages. I think this could be resolved if `tox` can create wheels and install them for each env.

**Comment by gaborbernat** [comment, marked as answer]
tox today supports a single package. You can create a plugin that supports multiple packages, but we don't plan to support it as part of the core for now. As you see here https://github.com/tox-dev/tox/blob/main/src/tox/tox_env/python/runner.py#L120C27-L120C68 we do allow for multiple packages, but the default packager here https://github.com/tox-dev/tox/blob/0fd872d64657b23220df0a9a1288a83bc11626ad/src/tox/tox_env/python/virtual_env/package/pyproject.py#L211-L267 only supports one.

**Reply by gaborbernat** [reply]
Should you create that plugin, please post it here 👍  for awareness of other people.

**Comment by tuukkamustonen** [comment]
I'm looking for how to use tox with `uv` [workspaces](https://docs.astral.sh/uv/concepts/projects/workspaces/) and ended up here.

`uv` will probably popularize "nested packages" / monorepos / whatever-you-want-to-call-it so I'd expect interest to grow.
